samples/excel/89-preview-apis/protect-data-in-worksheet-and-workbook-structure.yaml

```diff
@@ -0,0 +1,258 @@
+id: excel-protect-data-in-worksheet-and-workbook-structure
+name: Protect data in worksheet and the workbook structure
+description: Shows how to protect data in a worksheet and the workbook structure.
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#protect-data-in-worksheet").click(() => tryCatch(protectDataInWorksheet));
+        $("#unprotect-data-in-worksheet").click(() => tryCatch(unprotectDataInWorksheet));
+        $("#protect-workbook-structure").click(() => tryCatch(protectWorkbookStructure));
+        $("#unprotect-workbook-structure").click(() => tryCatch(unprotectWorkbookStructure));
+        $("#password-protect-data-in-worksheet").click(() => tryCatch(passwordProtectDataInWorksheet));
+        $("#password-unprotect-data-in-worksheet").click(() => tryCatch(passwordUnprotectDataInWorksheet));
+        $("#password-protect-workbook-structure").click(() => tryCatch(passwordProtectWorkbookStructure));
+        $("#password-unprotect-workbook-structure").click(() => tryCatch(passwordUnprotectWorkbookStructure));
+
+        async function protectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, true, false);
```

Comment:
My main concern is that you are nesting methods that call Excel.run inside methods that call Excel.run. It looks like the only purpose of the outer Excel.run is to get the context object. This seems like an unnecessary complication. If you keep the nested structure, is there a way you can rewrite the inner method so that it does not call Excel.run itself?